### PR TITLE
Support performing API with user token

### PR DESCRIPTION
**Pull request by ayalash**
Example for toml file content:
```
default = "production"

[servers.staging]
url = "https://platz.mydomain.com"
token = "..."
scheme = "XPlatzToken"

[servers.production]
url = "https://platz-production.mydomain.com"
token = "..."
scheme = "Bearer"
```

### CODE REVIEW

File: src/client/config.rs [hunk 1]
```diff
@@ -100,9 +172,13 @@ impl PlatzClientConfig {
 
     /// Returns Authorization header content, possibly refreshing the current
     /// credentials.
-    pub async fn get_authorization(&self) -> Result<String, PlatzClientError> {
+    pub async fn get_authorization(&self) -> Result<(String, String), PlatzClientError> {
```
Comment (popen2) [inline src/client/config.rs]: This works great, but if you want to be more strict you can use `reqwest`'s `HeaderName` and `HeaderValue` to ensure those values don't get mixed by callers.

File: src/client/error.rs [hunk 2]
```diff
@@ -1,11 +1,18 @@
 #[derive(Debug, thiserror::Error)]
 pub enum PlatzClientError {
-    #[error("Could not find any Platz config. Please set the PLATZ_URL and PLATZ_API_TOKEN environment variables. When running from inside a Platz deployment, mount the \"platz-creds\" secret to /var/run/secrets/platz.json .")]
+    #[error(r###"Could not find any Platz config. Use one of the following methods:
+    1. Set the PLATZ_URL and PLATZ_API_TOKEN environment variables.
+    2. Use "platz/config.toml" configuration file on your config directory.
```
Comment (popen2) [inline src/client/error.rs:5]:
```suggestion
    2. Use "~/.config/platz/config.toml" configuration file.
```

File: src/client/config.rs [hunk 3]
```diff
@@ -21,7 +38,9 @@ impl PlatzClientConfig {
     /// new_from_secret, whichever succeeds first.
     /// If no token can be found, a PlatzClientError::NotFound is returned.
     pub async fn new() -> Result<Self, PlatzClientError> {
-        if let Some(config) = Self::new_from_env()? {
+        if let Some(config) = Self::new_from_configuration(None).await? {
```
Comment (popen2) [inline src/client/config.rs]: Environment variables usually precede configuration files, I think the order should be env, then config files, and only then from secret.

**Comment by popen2** [comment]
```toml
[profile.staging]
url = "https://platz.mydomain.com"
user_token = "..."

[profile.production]
url = "https://platz-production.mydomain.com"
access_token = "..."
expires_at = "2023-02-19T08:03:20Z"
```

File: src/client/error.rs (hunk 2 above)
Comment (ayalash) [reply src/client/error.rs:5]: I don't think that `~/.config/platz/config.toml` will be correct, because it will be used in windows too.

**Comment by ayalash** [comment]
@popen2  I've pushed fixes for your CR comments (except for the one I answered), can you please re-review?

File: src/client/error.rs (hunk 2 above)
Comment (popen2) [reply src/client/error.rs:5]: Right, I forgot about windows...

I just think the error should include the actual paths.

Comment (popen2) [reply src/client/error.rs:5]: We'll improve it later.

**Pull request merged**